### PR TITLE
Create `!emojify`

### PR DESCRIPTION
**Pull request by bradleysigma**
Creates the `!emojify` command, that will convert argument text to emoji
image
For each character, will use each emoji an equal (±1) number of times.
Largely reimplments `reacitfy.js` from `unimplmeneted_commands`.

### CODE REVIEW

File: uqcsbot/scripts/emojify.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                   "html"]
+    master['6'] = [choice(["card-6-clubs", "card-6-diamonds", "card-6-hearts", "card-6-spades"])]
+    master['7'] = [choice(["card-7-clubs", "card-7-diamonds", "card-7-hearts", "card-7-spades"])]
+    master['8'] = [choice(["card-8-clubs", "card-8-diamonds", "card-8-hearts", "card-8-spades"]),
+                   "8ball"]
+    master['9'] = [choice(["card-9-clubs", "card-9-diamonds", "card-9-hearts", "card-9-spades"])]
+
+    master[' '] = ["whitespace"]
+
+    master['@'] = ["whip"]
```
Comment (nicklambourne) [inline uqcsbot/scripts/emojify.py]: Something for period and comma probably wouldn't go astray.

File: uqcsbot/scripts/emojify.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    # master['…'] = ["lastpass"]
+    # master['€'] = ["martian_euro"]
+    # master['√'] = ["sqrt"]
+    # master['∞'] = ["arduino", "visualstudio"]
+    # master['∴'] = ["julia"]
+
+    text = ""
+    if command.has_arg():
+        text = command.arg.upper()
+    text = text.replace("&AMP;", "&")
```
Comment (nicklambourne) [inline uqcsbot/scripts/emojify.py]: Why are these replacements necessary? Probably explain in a comment, too.

File: uqcsbot/scripts/emojify.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    master['&'] = ["ampersand", "dnd"]
+    master['*'] = ["day", "nab", "youtried", "msn_star", "rune_prayer"]
+    master['-'] = ["no_entry"]
+    master['+'] = ["tf2_medic", "flag-ch", "flag-england"]
+
+    master['"'] = [choice(["ldquo", "rdquo"]), "pig_nose"]
+    master['>'] = ["accenture", "implying", "plex", "powershell"]
+    master['/'] = ["slash"]
+
+    # master['Α'] = ["alpha"]
```
[the remark this answers is not in the document]
Comment (bradleysigma) [reply uqcsbot/scripts/emojify.py]: Somewhere these characters are converted to other characters (`Λ` to `L` etc., `…` to `...`, `€` to `EUR`, and `√`, `∞` & `∴` to `[?]`). Will be commented explicitly.

File: uqcsbot/scripts/emojify.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    emoji = {}
+    for c in set(text):
+        full, part = divmod(text.count(c), len(master[c]))
+        shuffle(master[c])
+        emoji[c] = full * master[c] + master[c][:part]
+        shuffle(emoji[c])
+
+    response = ""
+    for c in text:
+        response += ":"+emoji[c].pop()+":"
```
[the remark this answers is not in the document]
Comment (bradleysigma) [reply uqcsbot/scripts/emojify.py]: Fixed.

File: uqcsbot/scripts/emojify.py (hunk 1 above)
Comment (bradleysigma) [reply uqcsbot/scripts/emojify.py]: Fixed.

File: uqcsbot/scripts/emojify.py (hunk 2 above)
Comment (bradleysigma) [reply uqcsbot/scripts/emojify.py]: `&` converts to `&AMP;`, `>` converts to `&GT;`. This reverse that. Added comment.

File: uqcsbot/scripts/emojify.py (hunk 1 above)
Comment (bradleysigma) [reply uqcsbot/scripts/emojify.py]: `.` added. Can't find a good `,`.

File: uqcsbot/scripts/emojify.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+from random import shuffle, choice
+
+
+@bot.on_command("emojify")
+@loading_status
+def handle_emojify(command: Command):
+    '''
+    `!emojify text` - converts text to emoji.
+    '''
+    master: Dict[str, List[str]] = defaultdict(lambda: ["grey_question"])
```
[the remark this answers is not in the document]
Comment (bradleysigma) [reply uqcsbot/scripts/emojify.py]: Making it a util file only seems worth it if it were to be used by multiple commands, which seems unlikely to me.

**Pull request merged**